### DISCUSSION TITLE
Using azure key vault secrets as variables in config- csv file

### DISCUSSION THREAD

**Discussion by Git-Kamil** [Azure-Player/azure.datafactory.tools#180] [General]
Hi Kamil,

 I know that it is possible to use pipelines variables like $($Env:VARIABLE) in the config-csv file, but is it possible to use akv secrets as variable in the csv file?

**Comment by NowinskiK** [comment]
Yes, by adding an additional task for reading secrets from AKV. Just add PowerShell task or [Azure Key Vault task](https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/AzureKeyVaultV1/README.md) to read all required secrets and they will be available as environment variables within Agent.

**Reply by Git-Kamil** [reply]
Thanks Kamil,  I tried this today but it doesn't work. I need to do something wrong. I added a task to download a secret from akv. The secret name is called email. I the config-csv file I have  factory,adf-Test,globalParameters.emailaddress.value,"$($Env:EMAIL)" . Every time when I run it I receive the below error.
 ##[error]Cannot bind argument to parameter 'value' because it is an empty string.

**Reply by Git-Kamil** [reply]
Hi Kamil, 
I've resolved my issue. The error was caused by a wrong  value in the akv secret.  Thanks for your help.
